---
layout: default
title:  'Gymrek Lab Homepage'
---
<div class="row">
    <div class="col-xs-12">
	  	<p>
	    	<!-- bkgd image at top -->
	    	<center><img src="images/website-cover-fig.jpg" width="100%" class="img-responsive"></center>
	  	</p>
	</div>
</div>

<div class="row">

<p>
The Gymrek Lab is located at the University of California San Diego. We are part of the Department of Medicine's Division of Genetics and the Department of Computer Science and Engineering. We are also a member of the <a href="http://bioinformatics.ucsd.edu/">Bioinformatics and Systems Biology program</a> at UCSD.
</p>

<h2>What do we do?</h2>
Our overall goal is to understand complex genetic variants that underlie human disease. We are particularly interested in repetitive DNA variants known as short tandem repeats (STRs) as a model for complex variation. Our work focuses on developing computational tools for analyzing complex variation from large-scale sequencing data and applying these tools to learn about the contribution of repetitive variation to human disease. Another major interest of our group is understanding how repeats and other types of genetic variation impact gene regulation and other complex traits.

<table class="softwaretable" border="0" width=100% align="center">
  <tbody>
    <tr>
      <td width=33%>
        <center>
          <a href="http://webstr.ucsd.edu" target="_blank"><img width=65% src="images/icon_webstr.jpg"></a>
          <br>
          <a href="http://webstr.ucsd.edu" target="_blank">webstr.ucsd.edu</a>
        </center>
      </td>
      <td width=33%>
        <center>
          <a href="http://gymreklab.com/software.html" target="_blank"><img width=65% src="images/icon_code.jpg"></a>
          <br>
          <a href="http://gymreklab.com/software.html" target="_blank">Toolkit for STR Analysis</a>
        </center>
      </td>
      <td width=33%>
        <center>
          <a href="http://gymreklab.com/2018/03/05/snpstr_imputation.html" target="_blank"><img width=65% src="images/icon_imputation.jpg"></a>
          <br>
          <a href="http://gymreklab.com/2018/03/05/snpstr_imputation.html" target="_blank">STR Imputation</a>
        </center>
      </td>
    </tr>
  </tbody>
</table>

<h2>Featured publications</h2>
	
<table class="softwaretable" border="1" width=100% textalign="left">
  <tbody>
    <tr>
      <td width=25%>
        In this study, we develop an ensemble-based genotyping method <a href="https://github.com/gymrek-lab/EnsembleTR">(EnsembleTR)</a> and characterize ~1.7 million tandem repeats across 3,550 samples from 1000 Genomes and H3Africa projects.
      </td>
      <td width=75% bgcolor=#EEEEEE>
          <h4>A deep population reference panel of tandem repeat variation.</h4>
          Helyaneh Ziaei Jam, Yang Li, Ross DeVito, Nima Mousavi, Nichole Ma, Ibra Lujumba, Yagoub Adam, Mikhail Maksimov, Bonnie Huang, Egor Dolzhenko, Yunjiang Qiu, Fredrick Elishama Kakembo, Habi Joseph, Blessing Onyido, Jumoke Adeyemi, Mehrdad Bakhtiari, Jonghun Park, Sara Javadzadeh, Daudi Jjingo, Ezekiel Adebiyi, Vineet Bafna & Melissa Gymrek. (<em>Nature Communications</em> 2023). [<a href="https://www.nature.com/articles/s41467-023-42278-3">DOI</a>]
          <br>
          <img width=100% src="images/Website_Fig_EnsembleTR_Nature.jpg">
      </td>
    </tr>
  </tbody>
</table>
<br>	

<table class="softwaretable" border="1" width=100% textalign="left">
  <tbody>
    <tr>
      <td width=75% bgcolor=#EEEEEE>
          <h4>Patterns of de novo tandem repeat mutations and their role in autism.</h4>
          Ileena Mitra, Bonnie Huang, Nima Mousavi, Nichole Ma, Michael Lamkin, Richard Yanicky, Sharona Shleizer-Burko, Kirk E. Lohmueller, Melissa Gymrek. (<em>Nature </em> 2021). [<a href="https://doi.org/10.1038/s41586-020-03078-7">pubmed</a>]
          <br>
          <img width=100% src="images/Website_Fig_SSCDenovos_Nature.jpg">
      </td>
      <td width=25%>
        In this study we develop methods to identify (MonSTR) and prioritize (SISTR) de novo mutations at tandem repeats in WGS data, and then characterize patterns of TR mutations in the general healthy population, as well as, in individuals affected by autism. This work was a fun collaboration with the <a href="https://faculty.eeb.ucla.edu/Lohmueller/">Lohmueller Lab</a> from UCLA!
      </td>
    </tr>
  </tbody>
</table>
<br>	
<table class="softwaretable" border="1" width=100% textalign="left">
  <tbody>
    <tr>
      <td width=25%>
          We designed a machine learning framework, <a href="github.com/Pandaman-Ryan/AgentBind">AgentBind</a>, to identify and interpret sequence features that are most important for transcription factor (TF) binding. Unlike most previous works studying binding motifs, our work focuses on the sequence context in the vicinity of motifs and studies its role in TF binding.
      </td>
      <td width=75% bgcolor=#EEEEEE>
        <h4>Deep neural networks identify sequence context features predictive of transcription factor binding.</h4>
        An Zheng, Michael Lamkin, Hanqing Zhao, Cynthia Wu, Hao Su & Melissa Gymrek (<em>Nature Machine Intelligence</em> 2021). [<a href="https://doi.org/10.1038/s42256-020-00282-y" target="_blank">DOI</a>]
        <br>
        <img width=100% src="images/agentbind_fig.jpg">
      </td>
    </tr>
  </tbody>
</table>
<br>
<table class="softwaretable" border="1" width=100% textalign="left">
  <tbody>
    <tr>
      <td width=75% bgcolor=#EEEEEE>
        <h4>TRTools: a toolkit for genome-wide analysis of tandem repeats.</h4>
        Nima Mousavi, Jonathan Margoliash, Neha Pusarla, Shubham Saini, Richard Yanicky, Melissa Gymrek (<em>Bioinformatics</em> 2020). [<a href="https://pubmed.ncbi.nlm.nih.gov/32805020/" target="_blank">pubmed</a>]
        <br>
        <img width=100% src="images/trtools_fig.jpg">
      </td>
      <td width=25%>
          This manuscript describes <a href="github.com/gymreklab/trtools">TRTools</a>, a suite of tools for analysis of tandem repeat variation. It offers command-line tools for filtering, merging, and performing QC on VCFs containing tandem repeat genotypes. TRTools can also be imported as a Python library into custom scripts.
      </td>
    </tr>
  </tbody>
</table>
<br>
<table class="softwaretable" border="1" width=100% textalign="left">
  <tbody>
    <tr>
      <td width=25%>
        In this study we demonstrate that variation at thousands of tandem repeats across the genome have an impact on expression of nearby genes across many human tissue types. Effects are strongest for GC-rich STRs near transcription start sites. This work was a fun collaboration with the <a href="https://goren-lab.github.io/">Goren Lab</a> next door!
      </td>
      <td width=75% bgcolor=#EEEEEE>
          <h4>The impact of short tandem repeat variation on gene expression.</h4>
          Stephanie Feupe Fotsing, Jonathan Margoliash, Catherine Wang, Shubham Saini, Sharona Shleizer-Burko, Alon Goren, Melissa Gymrek (<em>Nature Genetics</em> 2019). [<a href="https://pubmed.ncbi.nlm.nih.gov/31676866/">pubmed</a>]
          <br>
          <img width=100% src="images/estrs_fig.jpg">
      </td>
    </tr>
  </tbody>
</table>
<br>
<table class="softwaretable" border="1" width=100% textalign="left">
  <tbody>
    <tr>
      <td width=75% bgcolor=#EEEEEE>
        <h4>Profiling the genome-wide landscape of tandem repeat expansions.</h4>
        Nima Mousavi, Sharona Shleizer-Burko, Richard Yanicky, Melissa Gymrek (<em>Nucleic Acids Research</em> 2019). [<a href="https://pubmed.ncbi.nlm.nih.gov/31194863/" target="_blank">pubmed</a>]
        <br>
        <img width=100% src="images/gangstr_fig.jpg">
      </td>
      <td width=25%>
          <a href="github.com/gymreklab/gangstr">GangSTR</a> is a tool for profiling tandem repeat variation from next-generation sequencing datasets. It takes as input aligned reads (BAMs) for one or more samples and outputs VCF files with estimated repeated lengths at a reference set of TRs. GangSTR can capture variation both at short and expanded TRs.
      </td>
    </tr>
  </tbody>
</table>
<br>
<table class="softwaretable" border="1" width=100% textalign="left">
  <tbody>
    <tr>
      <td width=25%>
        In this study we generate a SNP-STR haplotype panel by phasing STRs onto SNP haplotypes. This panel can be used to impute STR genotypes into datasets for which they cannot be directly genotyped (e.g. SNP genotyping array data commonly used for GWAS studies). We found that the majority of STRs can be accurately imputed, with a mean genome-wide imputation accuracy 97% for European samples.
      </td>
      <td width=75% bgcolor=#EEEEEE>
          <h4>A reference haplotype panel for genome-wide imputation of short tandem repeats.</h4>
          Shubham Saini, Ileena Mitra, Nima Mousavi, Stephanie Feupe Fotsing, Melissa Gymrek (<em>Nature Communications</em> 2018). [<a href="https://www.ncbi.nlm.nih.gov/pubmed/30353011" target="_blank">pubmed</a>]
          <br>
          <img width=100% src="images/imputation_fig.jpg">
      </td>
    </tr>
  </tbody>
</table>


</div>
