---
layout: default
title:  'Gymrek Lab - Research'
---

<h2>Research overview</h2>

Our overall goal is to understand genetic variants that underlie human disease and how their effects vary across different populations. We are a multidisciplinary lab and include both computational and wet lab biologists. We are particularly interested in repetitive DNA variants known as tandem repeats (TRs). Our work is often done in collaboration including with groups in the Depts. of Pediatrics, Medicine, Biomedical Informatics, Computer Science & Engineering, Electrical and Computer Engineering, Chemistry, and Psychiatry at UCSD as well as at other institutions. We are especially interested in building new collaborations with clinicians. We currently focus on the specific areas described below.

<table class="softwaretable">
<tbody>
<tr>
	<td width=30%>
		<img class="researchpic" src="images/research_figure1.jpeg" alt="research1">
	</td>
	<td width=70%>
		<h4>(1) Developing computational tools for analyzing complex variation in biobank-scale genomic datasets</h4>
		Tandem repeats (TRs) are some of the most polymorphic regions of the genome and make outsized contributions to disease but are technically challenging to analyze. Our lab develops methods to enable genome-wide analysis of TRs including:
		<ul>
			<li>Methods for genotyping TRs from short read (e.g. <a href="https://pubmed.ncbi.nlm.nih.gov/31194863/">GangSTR</a>, <a href="https://pubmed.ncbi.nlm.nih.gov/37872149/">EnsembleTR</a>) and long read (<a href="https://pubmed.ncbi.nlm.nih.gov/38965568/">LongTR</a>) whole genome sequencing datasets.</li>
			<li><a href="https://pubmed.ncbi.nlm.nih.gov/30353011/">Pipelines</a> and <a href="https://github.com/gymrek-lab/ensembletr?tab=readme-ov-file#version-iv-of-reference-snptr-haplotype-panel-for-imputation-of-tr-variants">reference panels</a> for imputing TRs from genotype data.</li>
			<li>Methods for performing TR-based <a href="https://pubmed.ncbi.nlm.nih.gov/38116119/">association studies</a>, <a href="https://pubmed.ncbi.nlm.nih.gov/39078205/">analysis of mosaicism</a>, and <a href="https://pubmed.ncbi.nlm.nih.gov/33442040/">analysis of de novo mutations</a> contributing to disease.</li> 
		</ul>
		<p> 
		Many of these tools have been applied to large datasets to gain new insights into patterns of TR variation and the contribution of TRs to different traits. Most of these tools including general utilities for filtering, QC, etc. of TR genotype data are packaged in our <a href="https://github.com/gymrek-lab/trtools">TRTools</a> package.
		</p>
		<p>
		TR genotypes generated by many studies we've been involved in are available on <a href="https://webstr.ucsd.edu/">WebSTR</a>, a site we built in collaboration with the Anisimova Lab at ZHAW.
		</p>
		<p>
		We are also interested in using pangenomes to understand genetic variation at repeats and other complex regions of the genome. Our lab is a member of the <a href="https://humanpangenome.org/">Human Pangenome Reference Consortium</a>. We are working on an <a href="https://pangenome.ucsd.edu/">interactive browser to visualize pangenome data</a>.
		</p>
		</td>
</tr>
<tr>
	<td width=30%>
		<img class="researchpic" src="images/Website_Fig_TRGWAS.jpg" alt="research2">
	</td>
	<td width=70%>
		<h4>(2) Identifying TRs contributing to human traits</h4>
		We have developed and applied methods to integrate TRs into association testing frameworks. We have applied these to uncover widespread contributions of TRs to a range of traits including:
		<ul>
			<li>Gene expression: we identified more than 1,000 high-confidence STRs acting as expression quantitative loci (eQTL) for nearby genes <a href="https://pubmed.ncbi.nlm.nih.gov/31676866/">(Fotsing et al 2019)</a>. We previously estimated that STRs contribute ~10-15% of the cis heritability of gene expression in humans (<a href="https://pubmed.ncbi.nlm.nih.gov/26642241/">Gymrek et al 2016)</a>.</li>
			<li>Blood and serum traits: we identified 119 candidate causal STR-trait associations and estimate that STRs account for 5.2%â€“7.6% of causal variants identifiable from GWAS from blood biomarkers.</li>
		</ul>
		<p>
			We have also developed and applied methods to study de novo mutations at STRs, which we identified as contributing to risk for autism spectrum disorders (<a href="https://pubmed.ncbi.nlm.nih.gov/33442040/">Mitra et al. 2021</a>). We are continuing to apply our association testing and de novo analysis frameworks to study the contribution of TRs to other traits including molecular and disease phenotypes. We also have multiple ongoing collaborative projects to perform genome-editing of predicted pathogenic TRs in human iPSCs and other cell types.
		</p>
	</td>
</tr>
<tr>
	<td width=30%>
		<img class="researchpic" src="images/wu_cover.jpg" alt="research3">
	</td>
	<td width=70%>
		<h4>(3) Studying mutation and selection processes at TRs within and across species</h4>
		Tandem repeats have multiple interesting properties compared to other types of variation, including rapid mutation rates and high rates of multi-allelicness. Understanding the evolutionary forces including mutation and selection driving patterns of variation at these loci is critical to predicting which TR mutations are likely to be pathogenic. We have made multiple contributions including:
		<ul>
			<li>Developing mathematical models of TR mutation that enable inference of mutation rates and other parameters at individual loci <a href="https://pubmed.ncbi.nlm.nih.gov/28892063/">(Gymrek et al. 2017)</a></li>
			<li>Developing methods (<a href="https://pubmed.ncbi.nlm.nih.gov/33442040/">SISTR</a> and <a href="https://www.biorxiv.org/content/10.1101/2022.05.12.491726v1">SISTR2</a>) to infer negative selection at TRs</li>
			<li>Identifying Msh3 as a modifier of the rate of genome-wide TR expansions in in bred mice <a href="https://pubmed.ncbi.nlm.nih.gov/37127331/">(Maksimov & Wu et al. 2023)</a> (highlighted on the cover of Genome Research!).</li>
		</ul>
		We are currently working on expanding our models of selection to enable inference in non-European populations and modeling TR mutation/selection both within and across species.
	</td>
</tr>

<tr>
	<td width=30%>
		<img class="researchpic" src="images/research_fig4.jpg" alt="research4">
	</td>
	<td width=70%>
		<h4>(4) Understanding how the effects genetic variants differ across human populations</h4>
		Together with Drs. Kelly Frazer (UCSD) and Lucila Ohno-Machado (Yale), we lead the <a href="https://admixgenomics.ucsd.edu/">Center for Admixture Science and Technology</a> (CAST), which focuses on improving the utility of genomics methods for admixed populations. Through CAST we work on multiple projects including:
		<ul>
			<li>Admixture methods and other techniques to understand the contribution of local ancestry to complex traits.</li>
			<li>Methods for fine-mapping in multi-ancestry settings (e.g. <a href="https://www.medrxiv.org/content/10.1101/2025.11.13.25339614v1">PIPSORT</a>) and to incorporate complex ancestry-specific variants and haplotypes</li>
			<li>Machine learning methods to model social determinants together with genetic risk factors for disease.</li>
		</ul> 
	</td>
</tr>

<tr>
	<td width=30%>
		<img class="researchpic" src="images/research_fig5.jpg" alt="research5">
	</td>
	<td width=70%>
		<h4>(5) Using high-throughput experimental techniques to study the impacts of genetic variants on molecular and cellcular phenotypes</h4>
		We have multiple collaborative projects to study the impact of genetic variants in human cells including:
		<ul>
			<li>Using massively parallel reporter assays (MPRA) to understand the effects of TRs on gene regulation (collaboration with Goren Lab, see our <a href="https://www.biorxiv.org/content/10.1101/2025.09.14.676153v1">preprint</a>!)</li>
			<li>Using base editing to study mutations in genes implicated in rare disease (collaboration with Goren and Komor Labs)</li>
		</ul>
	</td>
</tr>

</tbody>
</table>

